### PR TITLE
Fix malformed download responses for dev installs

### PR DESCRIPTION
**Pull request by alikins**
The StreamingHttpResponse() needs an iterable, so
use Requests response.iter_content(). Using
response.content directly doesn't work.

Fixes: ansible/galaxy-dev#117

After the change, `ansible-galaxy collection install` to a local galaxy/ah will work.

Direct http examples after change:

```
(ansible-py3) [newswoop:F29:ansible (devel % u=)]$ http --verbose --download http://localhost:5001/api/automation-hub/v3/artifacts/collections/alikins.collection_ntp.0.1.182
GET /api/automation-hub/v3/artifacts/collections/alikins.collection_ntp.0.1.182 HTTP/1.1
Accept: */*
Accept-Encoding: identity
Connection: keep-alive
Host: localhost:5001
User-Agent: HTTPie/0.9.9



HTTP/1.1 200 OK
Allow: GET, HEAD, OPTIONS
Connection: close
Content-Type: application/octet-stream
Date: Thu, 19 Sep 2019 18:48:59 GMT
Server: WSGIServer/0.2 CPython/3.6.8
Vary: Accept
X-Frame-Options: SAMEORIGIN

Downloading to "alikins.collection_ntp.0.1.182-7"
Done. 8.01 kB in 0.29746s (26.93 kB/s)
```

```
$ file alikins.collection_ntp.0.1.182-7
alikins.collection_ntp.0.1.182-7: gzip compressed data, was "alikins-collection_ntp-0.1.182.tar", last modified: Tue May 14 17:13:11 2019, max compression, original size 61440
```

```
$ tar tvfl alikins.collection_ntp.0.1.182-7
tar: Removing leading `/' from member names
drwxrwxr-x adrian/adrian     0 2019-05-08 12:14 /
drwxrwxr-x adrian/adrian     0 2019-04-02 10:41 roles/
-rw-rw-r-- adrian/adrian   168 2019-04-18 13:51 =345.43.432
-rw-rw-r-- adrian/adrian    21 2019-04-02 12:39 README.md
-rw-rw-r-- adrian/adrian  1080 2019-04-02 10:45 LICENSE
-rw-rw-r-- adrian/adrian   294 2019-05-14 13:13 galaxy.yml
...
```

### CODE REVIEW

File: galaxy_api/api/v3/viewsets.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def get(self, request, *args, **kwargs):
 
         if response.status_code == requests.codes.ok:
             return StreamingHttpResponse(
-                response.content,
+                response.iter_content(),
```
Comment (alikins) [inline galaxy_api/api/v3/viewsets.py]: May make sense to use chunk_size=None here
``` python
response.iter_content(chunk_size=None)
```
 based on https://2.python-requests.org/en/master/api/#requests.Response.iter_content:

> chunk_size must be of type int or None. A value of None will function differently depending on the value of stream. stream=True will read data as it arrives in whatever size the chunks are received. If stream=False, data is returned as a single chunk.

Comment (alikins) [reply galaxy_api/api/v3/viewsets.py]: Yeah, definitely needs chunk_size=None or larger than one. The default chunk_size=1 causes small ish downloads
to be pretty slow. In an all localhost test, default chunk_size=1 for ~200kb to avg about 7 seconds and ~26kb/s:

```
Downloading to "alikins.k8s_demo_collection.0.37.3-1"
Done. 192.72 kB in 7.32279s (26.32 kB/s)
```

And chunk_size=None, downloads average about 0.001 seconds and 159MB/s (localhost in full effect...)

```
Downloading to "alikins.k8s_demo_collection.0.37.3"
Done. 192.72 kB in 0.00121s (154.99 MB/s)
```

**Comment by alikins** [comment]
Some more non-scientific testing seems to suggest a chunk_size of 4096 or 8192 would likely be sufficient for local/dev installs (my all-localhost setup matched the 150 MB/s numbers with 8k buffers). But then, for dev machines, with lots of bandwidth, memory, and limited concurrency chunk_size=None likely makes sense. For potentially standalone/private instances, the chunk_size may need to be a tunable though.

**Comment by cutwater** [comment]
@alikins Can you please set chunk_size to 4096.

RCA: The problem is using  `.content` attribute. `Response.__iter__` is indeed calling `iter_content()`, but it's different from the `.content` attribute. It should've been `StreamingHttpResponse(response, ...)` initially.

**Pull request merged**